### ISSUE TITLE
providers:registered without fire

### ISSUE THREAD

**Issue by TungCena** [adonisjs/adonis-framework#555]
bootstrap\events.js
```  
const Ioc = require('adonis-fold').Ioc

Ioc.on('providers:registered', () => {
  // all providers have been registered
})
```
It cannot be fired , how can I fire all providers have been registered event? thanks!

**Comment by RomainLanz** [comment]
Hey @TungCena 👋🏼

This event is fired by the `IoC` itself.

**Comment by TungCena** [comment]
Hi @RomainLanz ,
I have two providers, I want to call a function when these providers have been registered

**Comment by RomainLanz** [comment]
For that you'll need to use the `bind` function of your provider.

```js
const ServiceProvider = require('adonis-fold').ServiceProvider

class MyServiceProvider extends ServiceProvider {

  * register () { 
    // Register this Service Provider
  }

  * boot () { 
    // All Service Provider are registered.
    // Do some hard work.
  }

}
```

- [IoC Container & Service Providers ](http://adonisjs.com/docs/3.2/ioc-container#_service_providers)

**Comment by TungCena** [comment]
```
class A extends ServiceProvider {

  * register () { 
        this.app.singleton('A', (app) => {
              .............
       }) 
  }
}
```

```
class B extends ServiceProvider {

  * register () { 
        this.app.singleton('B', (app) => {
         ............
       })
  }

  * boot (){
     const A = use('A')
    //throw error may be A is not binded
  }
}

**Comment by thetutlage** [comment]
@TungCena How to register the providers. Registering `A` and `B` together will never have any issues and you can access the `A` inside the `B` boot method

**Comment by TungCena** [comment]
`providers/aProvider.js` to register a, 
`providers/bProvider.js` to register b, 

```
class B extends ServiceProvider {

  * register () { 
        this.app.singleton('B', (app) => {
         ............
       })
  }

  * boot (){
     const A = use('A')
     //Error: Cannot find module 'config/app.js'
  }
}
```

```
class A extends ServiceProvider {

  * register () { 
        this.app.singleton('A', (app) => {
          const Config = app.use('Adonis/Src/Config')
          return new SomeThing(Config.config.app.something);
       }) 
  }
}
```

**Comment by thetutlage** [comment]
I tried your code and it works fine, just one change. I have move the `const A = use('A')` inside an event, since it is dependent upon `Config` provider which needs to know the config directory path, which is set after `bind:autoload` event.

So following inside the `B` provider `boot` method should work

```js
    this.app.once('bind:autoload', () => {
      const A = use('A')
    })
``` 

I will remove this in future versions of Adonis to make sure it's not that complicated

**Comment by thetutlage** [comment]
Fixed in 4.0, no need to listen for `bind:autoload` event.
